### ISSUE TITLE
404 instead of 500 when url is not found

### ISSUE THREAD

**Issue by calidion** [balderdashy/sails#2218]
currently when sails encounters a url mismatch it would invoke adapter and give an 500 server error.
that seems not correct.
if the url is not found in side the controller, an HTTP 404 would be acceptable.

current message is : 

```
error: Sending 500 ("Server Error") response: 
 Error (E_UNKNOWN) :: Encountered an unexpected error
: ER_BAD_FIELD_ERROR: Unknown column 'NaN' in 'where clause'
```

**Comment by loicsaintroch** [comment]
Do you still have this issue @calidion @caasjj?

**Comment by CWyrtzen** [comment]
Is this still happening with the latest release? v0.11

**Comment by kevinburkeshyp** [comment]
might be a dupe of https://github.com/balderdashy/sails/issues/2829 (which I was able to repro with sails 0.11)

**Comment by sailsbot** [comment]
Thanks for posting, @calidion.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
